homeassistant/components/mqtt/vacuum/__init__.py

```diff
@@ -27,26 +30,35 @@
 )
 
 
-def validate_mqtt_vacuum_discovery(value):
+def validate_mqtt_vacuum_discovery(config_value: ConfigType) -> ConfigType:
     """Validate MQTT vacuum schema."""
-    schemas = {LEGACY: DISCOVERY_SCHEMA_LEGACY, STATE: DISCOVERY_SCHEMA_STATE}
-    return schemas[value[CONF_SCHEMA]](value)
+    schemas: dict[str, vol.Schema] = {
+        LEGACY: DISCOVERY_SCHEMA_LEGACY,
+        STATE: DISCOVERY_SCHEMA_STATE,
+    }
```

Comment:
Let this be inferred.
```suggestion
    schemas = {LEGACY: DISCOVERY_SCHEMA_LEGACY, STATE: DISCOVERY_SCHEMA_STATE}
```